### PR TITLE
[SPARK-22816][TEST] Basic tests for PromoteStrings and InConversion

### PR DESCRIPTION
**Pull request by wangyum**
## What changes were proposed in this pull request?

Test Coverage for `PromoteStrings` and `InConversion`, this is a Sub-tasks for [SPARK-22722](https://issues.apache.org/jira/browse/SPARK-22722).

## How was this patch tested?

N/A

### CODE REVIEW

File: sql/core/src/test/resources/sql-tests/inputs/typeCoercion/native/inConversion.sql [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+SELECT cast(1 as string) in (cast(1 as int)) FROM t;
+SELECT cast(1 as string) in (cast(1 as bigint)) FROM t;
+SELECT cast(1 as string) in (cast(1 as float)) FROM t;
+SELECT cast(1 as string) in (cast(1 as double)) FROM t;
+SELECT cast(1 as string) in (cast(1 as decimal(10, 0))) FROM t;
+SELECT cast(1 as string) in (cast(1 as string)) FROM t;
+SELECT cast(1 as string) in (cast('2017-12-11 09:30:00.0' as timestamp)) FROM t;
+SELECT cast(1 as string) in (cast('2017-12-11 09:30:00' as date)) FROM t;
+-- Hive and Spark SQL doesn't support below operations:
+-- SELECT cast(1 as string) in (cast('1' as binary)) FROM t;
```
Comment (gatorsmile) [inline sql/core/src/test/resources/sql-tests/inputs/typeCoercion/native/inConversion.sql]: For the cases we do not support, we should still keep them. 

Just ensure the type coercion rule is not changed and the error messages are reasonable.

**Comment by HyukjinKwon** [comment]
retest this please

**Comment by gatorsmile** [comment]
LGTM

**Comment by gatorsmile** [comment]
Thanks! Merged to master.

**Pull request closed without merging**